src/sdk/pynni/nni/compression/torch/compressor.py

```diff
@@ -87,26 +98,26 @@ def compress(self):
         torch.nn.Module
             model with specified modules compressed.
         """
-        if self.modules_wrapper is not None:
-            # already compressed
-            return self.bound_model
-        else:
-            self.modules_wrapper = []
-
-        modules_to_compress = self.detect_modules_to_compress()
-        for layer, config in modules_to_compress:
-            wrapper = self._wrap_modules(layer, config)
-            self.modules_wrapper.append(wrapper)
-
-        self._wrap_model()
         return self.bound_model
 
     def register_buffer(self, name, value):
         """
-        To register buffers used in wrapped module's forward method.
+        To register buffers or add regular variable used in wrapped module's forward method.
+        If the type of the value is Torch.tensor, then this value is registered as a buffer in wrapper,
+        which will be saved by model.state_dict. Otherwise, this value is just a regular variable in wrapper.
 
+        Parameters
+        ----------
+        name : str
+            name of the variable
+        value: any
+            value of the variable
         """
-        self.buffers[name] = value
+        for wrapper in self.get_modules_wrapper():
+            if isinstance(value, torch.Tensor):
+                wrapper.register_buffer(name, value.clone())
```

Comment:
1. Since wrapper's masks already registered within wrapper, it seems there is no requirement for now to register a tensor buffer?
2. It is convenient to use one method to set an attribute to all wrappers, maybe change this api to another name?

Reply:
rename it to register variable now.

Reply:
rename it to set_wrappers_attribute